### DISCUSSION TITLE
FW 0.9.48 (série E) control_function switch KO

### DISCUSSION THREAD

**Discussion by Clust3r** [Arnold-n/P1P2MQTT#97] [General]
Hi i just upgrade from 0.9.43rc to 0.9.48 live the separated topics but I loose my control under HA for DHW boost etc.

When i try to active the control_function switch. Nothing change. 

![IMG_0888](https://github.com/Arnold-n/P1P2MQTT/assets/249758/92d09cd7-63e1-452f-9db0-b2bd17130f1a)

**Comment by Arnold-n** [comment]
Thanks for notifying. One of the configuration messages is truncated, which may explain your control loss. P1P2MQTT-bridge v0.9.49 for the ESP increases the maximum configuration message size to 1500 and should solve this.

I am somewhat surprised about the longer msg size on your side, if you don't mind, could you (after upgrade) share the (longest) homeassistant/# message(s)? Note: If you changed any of the predefined names to personal ones, these may be visible in the config messages.

**Reply by Clust3r** [reply]
Thanks for the reply,

Just upgrade to the 0.9.50 FW, the message about the too long config is now fixed. I'm not sure but the only new entity found after the upgrade was:
P1P2/M/P1P2MQTT/bridge0/A  (String of 1025 like the B C or D one very strange)

But still can't get control access:
`[ESP] 2024-05-19 13:01:09 Throttling at 4
[ESP] 2024-05-19 13:01:10 Throttling at 3
[ESP] 2024-05-19 13:01:11 Throttling at 2
[ESP] 2024-05-19 13:01:12 Throttling at 1
[ESP] 2024-05-19 13:01:13 Ready throttling
[MON] 2024-05-19 13:01:56 CONTROL_ID is 0xF0, changing control mode to L1
`

Nothing after this line and still no control under HA 
<img width="776" alt="image" src="https://github.com/Arnold-n/P1P2MQTT/assets/249758/bf9d9f31-bfb8-486f-bcfe-324ed0d94ea6">

**Reply by Arnold-n** [reply]
Thanks. The P1P2/M/# messages are merely used to store data over an ESP reset or power-down - the ESP reads back this data when restarted, to reduce the time it takes to re-acquire all data from the interface and re-communicate it to HA.

I cannot test DHW_Boost as I do not have a tank for DHW myself. Your system seems to operate nicely in auxiliary control mode, i.e., it communicates with the room thermostat. Can you see if  `Control_Function` switch in HC_Bridge is on? If off, DHW controls remain unavailable. If on, does any of the other controls work (setting Altherma-On on/off, setting LWT setpoint)? If yes, can you manually set the J-mask to 0x10B (`J12B` - corrected by edit), wait a few minutes until telnet becomes quiet again, then on the room thermostat change the boost status manually and check any output via telnet?

**Reply by Clust3r** [reply]
Thanks again to complete my previous post, 2 things

1 : When i click "Activate" on the Control_Function switch the Switch remain in "Unknown" state

<img width="479" alt="image" src="https://github.com/Arnold-n/P1P2MQTT/assets/249758/90fabe63-ab40-4552-80b9-fafe65ae95a4">

2: I take DHT for example but all Swith are disabled:
<img width="262" alt="image" src="https://github.com/Arnold-n/P1P2MQTT/assets/249758/a111854e-82ad-4f62-b01a-d76092826477">


I'll try the J-Mask.

Thanks

**Reply by Arnold-n** [reply]
Sorry, for viewing data via telnet the J-mask must be J12B (or you need to log P1P2/P/#).

**Reply by Arnold-n** [reply]
But if all switches remain unavailable something else goes wrong. Did you rebuild the MQTT topics via the MQTT_Rebuild button (or telnet `D3` command) (necessary after a firmware upgrade, I should make that automatic)?

**Reply by Clust3r** [reply]
Yes allready do it, just redo to be sure. 

System operate normaly nothing strange etc. But still can't get any "switch" active.

Debuging via J-Mask doesn't give me any advise or something to investigate.

**Reply by Arnold-n** [reply]
Just to be sure: did you also update P1P2Monitor on the ATmega to v0.9.46? If so, could you share (perhaps by mail) a full log of MQTT traffic (#) during and after a reboot (`D0` and MQTT rebuild `D12`)?

**Reply by Clust3r** [reply]
Yes ATmega allready updated:

`[ESP] 2024-05-19 17:16:07 Subscribed to P1P2/W result 5
[ESP] 2024-05-19 17:16:07 Subscribed to P1P2/W/P1P2MQTT/bridge0 result 6
[ESP] 2024-05-19 17:16:07 Subscribed to homeassistant/status
[ESP] 2024-05-19 17:16:07 Subscribed to P1P2/P/meter/U/9/Electricity_Power
[ESP] 2024-05-19 17:16:07 Subscribed to P1P2/P/meter/U/9/Electricity_Total
[ESP] 2024-05-19 17:16:07 Subscribed to P1P2/P/meter/U/9/Electricity_BUH_Power
[ESP] 2024-05-19 17:16:07 Subscribed to P1P2/P/meter/U/9/Electricity_BUH_Total
[ESP] 2024-05-19 17:16:07 < ESP reboot reason: D0
[ESP] 2024-05-19 17:16:07 < No ethernet hardware detected
[MON] 2024-05-19 17:16:07 P1P2Monitor v0.9.46 compiled May 18 2024 15:53:07 bran                                                                                                                                                                                                                                                                                                                                                           d 1 modelnr 3 +control NEWP1P2LIB E-series
[ESP] 2024-05-19 17:16:07 < Connected to WiFi
[MON] 2024-05-19 17:16:07 Reset cause: MCUSR=5 (brown-out) (power-on-reset)
[ESP] 2024-05-19 17:16:07 < MQTT client connected
[MON] 2024-05-19 17:16:07 P1P2MQTT bridge ATmegaHwID v1.2
[MON] 2024-05-19 17:16:07 Control_id=0xF0
[ESP] 2024-05-19 17:16:07 < Telnet 10.0.10.191 connected
[MON] 2024-05-19 17:16:07 Counter_requests=1
`

Any advise to log all MQTT traffic ? I'm not very familiar with MQTT loging mecanics

**Reply by Clust3r** [reply]
... I'm sorry just find the "solution" i have made intervention on my LWT circuit 2 days ago and the system was still in "Installer mode" (Code 5678) go back to "Advanced User Mode" Code 1234 

got back all the switches 

![image](https://github.com/Arnold-n/P1P2MQTT/assets/249758/506ff9ce-3fae-40c5-b4fa-33691f953a59)

Sorry for the time lost on this.
Regards

**Reply by Arnold-n** [reply]
No problem at all - this is crucial information for others. My system just continues to operate in installer mode (and drops out of installer mode after an hour). It is interesting to learn that installer mode blocks regular control from a  second controller in your system. I'd be interested to know how we can see in the hex data logs that the system is in installer mode, so we can add feedback in HA - this may be useful also for the quiet level discussion.

**Reply by Clust3r** [reply]
If i can help no problem. I can switch User Mode and monitoring via telnet. ATM tried to but when i change the user mode from User to installer, i didn't found any change in the log at the same time. 

Maybe i need to use/dump something specific ? 

Regards

**Reply by Arnold-n** [reply]
Perhaps there is information in packet 0x30 in the raw hex data, which is not decoded into MQTT entities currently. If you could log P1P2/R/# data from telnet (`J012`) when making the user/installer change or back, that would be helpful.
